File: software/examples/java/ExampleSleep.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickHAT;

public class ExampleSleep {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XXYYZZ to the UID of your HAT Brick
	private static final String UID = "XXYYZZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickHAT hat = new BrickHAT(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Turn Raspberry Pi and Bricklets off in 2 seconds for 30 minutes with sleep indicator enabled
		hat.setSleepMode(2000, 1800000, true, true, true);

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/javascript/ExampleSleep.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XXYYZZ'; // Change XXYYZZ to the UID of your HAT Brick

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var hat = new Tinkerforge.BrickHAT(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Turn Raspberry Pi and Bricklets off in 2 seconds for 30 minutes with sleep indicator enabled
        hat.setSleepMode(2, 1800, true, true, true);
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
